src/app/work-item/work-item.service.ts

```diff
@@ -581,13 +586,25 @@ export class WorkItemService {
         let updatedWorkItem = response.json().data as WorkItem;
         // Find the index in the big list
         let updateIndex = this.workItems.findIndex(item => item.id == updatedWorkItem.id);
+        //Item is in the list
         if (updateIndex > -1) {
-          // Update work item attributes
-          this.workItems[updateIndex].attributes = updatedWorkItem.attributes;
-          this.workItems[updateIndex].relationships.baseType = updatedWorkItem.relationships.baseType;
-          // Resolve users for the updated item
-          this.resolveUsersForWorkItem(this.workItems[updateIndex]);
+          if (this.doesMatchCurrentFilter(updatedWorkItem)) {
+            // Update work item attributes
+            this.workItems[updateIndex].attributes = updatedWorkItem.attributes;
+            this.workItems[updateIndex].relationships.assignees = updatedWorkItem.relationships.assignees;
+            this.workItems[updateIndex].relationships.creator = updatedWorkItem.relationships.creator;
+            this.workItems[updateIndex].relationships.baseType = updatedWorkItem.relationships.baseType;
+            // Resolve users for the updated item
+            this.resolveUsersForWorkItem(this.workItems[updateIndex]);
+          } else {
+            // Remove the item from the list
+            this.workItems.splice(updateIndex, 1);
```

Comment:
pop would remove the last element. We are using splice as the element could be anywhere on the list.